Sources/BuildableMacros/BuildableTracked/BuildableTrackedMacro.swift

```diff
@@ -40,39 +40,27 @@ public struct BuildableTrackedMacro: PeerMacro {
     }
 
     private static func lowestAccessLevelModifier(for decl: VariableDeclSyntax) -> DeclModifierSyntax? {
-        var syntax: DeclModifierSyntax?
         if decl.modifiers.isEmpty { return nil }
 
-        let modifiers = decl.modifiers.lazy.map(\.name.text)
-        if modifiers.contains("private") {
-            syntax = DeclModifierSyntax(name: .keyword(.private))
-        } else if modifiers.contains("fileprivate") {
-            syntax = DeclModifierSyntax(name: .keyword(.fileprivate))
-        } else if modifiers.contains("internal") {
-            syntax = DeclModifierSyntax(name: .keyword(.internal))
-        } else if modifiers.contains("package") {
-            syntax = DeclModifierSyntax(name: .keyword(.package))
-        } else if modifiers.contains("public") {
-            syntax = DeclModifierSyntax(name: .keyword(.public))
-        } else if modifiers.contains("open") {
-            syntax = DeclModifierSyntax(name: .keyword(.open))
-        } else {
-            syntax = nil
-        }
+        let modifiers = decl.modifiers.lazy.map(\.name)
+        guard let name = modifiers.last(where: {
+            guard case let .keyword(accessLevel) = $0.tokenKind else { return false }
+            let accessLevels: [SwiftSyntax.Keyword] = [.private, .fileprivate, .internal, .package, .public, .open]
+            return accessLevels.contains(accessLevel)
+        }) else { return nil }
```

Comment:
This logic is based on the order of the modifiers which is not correct.

Consider the following:

```swift
public struct Sample {
    @BuildableTracked
    private(set) public var p1: String
    @BuildableTracked
    public private(set) var p2: String
}
```

Macro expands to:

```swift
struct Sample {
    private(set) public var p1: String
    public func p1(_ value: String) -> Self {
        var copy = self
        copy.p1 = value
        return copy
    }
    public private(set) var p2: String
    private func p2(_ value: String) -> Self {
        var copy = self
        copy.p2 = value
        return copy
    }
}
```

Although p1 and p2 have exactly the same access level, this logic is not able to handle it properly.

Reply:
@MojtabaHs Can you please check this review? It would be great if you add a test case for this specific scenario. Thanks. 🙏

Reply:
I suspected that when I used `last` in the refactored code. It was originally `first` and I had to change it to `last` to pass the tests and match with the original intention. Since it is a refactor PR, I don't see any changes in the original logic and I think it needs another PR.

Are you sure about the original logic? Maybe we should add a test on the original, make it green and come back to this refactor PR later. What do you think?

Reply:
If this PR is match with the original code, we can merge and continue on [this PR](https://github.com/Mr-Alirezaa/BuildableMacro/pull/12) for to fix this issue.

Reply:
> Then we can continue on https://github.com/Mr-Alirezaa/BuildableMacro/pull/12 for to fix this issue.

I do agree with this. Let's decide on the logic first on #12.